adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -698,6 +703,81 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
   }
 
   /**
+   * This method gets the sort and partition map metadata from the header of the file
+   * give as input.
+   *
+   * @param filename the filename for the metadata
+   * @return a partition map if the data was written sorted, or an empty Seq if unsorted
+   */
+  private[rdd] def extractPartitionMap(filename: String): Seq[(ReferenceRegion, ReferenceRegion)] = {
+    // the sorted metadata is always stored with the sequence dictionary metadata
+    val path = new Path(filename + "/_seqdict.avro")
+    val fs = path.getFileSystem(sc.hadoopConfiguration)
+
+    try {
+      // get an input stream
+      val is = fs.open(path).asInstanceOf[InputStream]
+
+      // set up avro for reading
+      val dr = new GenericDatumReader[GenericRecord]
+      val fr = new DataFileStream[GenericRecord](is, dr)
+
+      // parsing the json from the metadata header
+      // this unfortunately seems to be the only way to do this
+      // avro does not seem to support getting metadata fields out once
+      // you have the input from the string
+      val metaDataMap = JSON.parseFull(fr.getMetaString("avro.schema"))
```

Comment:
I was thinking of adding reference region to bdg-formats for other reasons, and extending it a bit to make it more Guava Range-ish:
```
enum BoundType { CLOSED, OPEN }

record ReferenceRegion {
  string name;
  union { BoundType, null } lowerBound = CLOSED;
  long lower;
  union { BoundType, null} upperBound = OPEN;
  long upper;
}
```

I find this potentially useful since as we use `[closed, open)` ranges elsewhere in bdg-formats, there are "genomic algebra" operations that might want to use other bound types, including e.g. `[closed, null` to represent open-ended ranges.

Could a map of these records be serialized to an avro file similar to how sequence dictionaries and samples are?